src/nvme/mi.c

```diff
@@ -327,7 +327,7 @@ int nvme_mi_scan_ep(nvme_mi_ep_t ep, bool force_rescan)
 
 	rc = nvme_mi_mi_read_mi_data_ctrl_list(ep, 0, &list);
 	if (rc)
-		return -1;
+		return rc;
```

Comment:
The library uses the errno style error handling approach. So the return code `-1` is correct. But I see that `nvme_mi_mi_read_mi_data_ctrl_list` is not following this approach. Thus I think you need to update that code.

Reply:
Indeed :)